plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -241,16 +250,16 @@ def pod_statuses(pods, release_doc)
       end.each(&:check)
 
       # If a role is autoscaled, there is a chance pods can be deleted during a deployment.
-      # Sort them by "most alive" and use the first one, so we ensure at least one pods works.
+      # Sort them by "most alive" and use the min ones, so we ensure at least that number of pods work.
       if role.autoscaled?
         statuses.sort_by! do |status|
           if status.live
             -1
           else
             status.finished ? 1 : 0
           end
-        end.slice!(1..-1)
-        statuses.each { |s| s.details += " (autoscaled role, only showing one pod)" }
+        end
+        statuses = statuses.first(release_doc.desired_pod_count)
```

Comment:
what is the desired count for an autoscaled role? couldn't it be higher than the HPA "minimum"?

Reply:
if it's higher then your deploy will fail ... that will be addressed once we keep them in sync
before it was only verifying 1 pod, the new plan is the verify minimum-pods